nixos/modules/services/networking/gns3-server.nix

```
@@ -227,14 +233,25 @@ in {
         User = "gns3";
         WorkingDirectory = "%S/gns3";
 
+        # Required for ubridge integration to work
+        #
+        # GNS3 needs to run SUID binaries (ubridge)
+        # but NoNewPrivileges breaks execution of SUID binaries
+        DynamicUser = lib.mkForce false;
+        NoNewPrivileges = lib.mkForce false;
+        RestrictSUIDSGID = lib.mkForce false;
+        PrivateUsers = lib.mkForce false;
+
         # Hardening
-        DeviceAllow = lib.optional flags.enableLibvirtd "/dev/kvm";
+        DeviceAllow = [
+          "/dev/net/tap rw" # ubridge needs access to tap devices
```

```suggestion
          # ubridge needs access to tap and tun devices
          "/dev/net/tap rw"
          "/dev/net/tun rw"
```
Looks like you need tun as well, otherwise I still get the following error:
```
Error while sending command 'bridge add_nio_tap bridge0 tap-gns3-e0': unable to create NIO TAP for bridge 'bridge0': uBridge version 0.9.18 running with libpcap version 1.10.4 (with TPACKET_V3)
Hypervisor TCP control server started (IP 0.0.0.0 port 37651).
create_nio_tap: unable to open TAP device tap-gns3-e0 (Operation not permitted)
```

Interesting enough, on my device it still works with only tun, and without tap.
Either the error message is wrong, or something else is going on behind scene, my networking knowledge is not deep enough.

---

@dani0854 Thank you for the review!

Oops sorry, I forgot to force push the commit adding `/dev/net/tun rw` to `DeviceAllow`.

`/dev/net/tun rw` is required for ubridge: https://github.com/GNS3/ubridge/blob/2f7798c48e718b44e210d16cf0592dbaa085e9be/src/nio_tap.c#L67